orca_detector/database_parser.py

```diff
@@ -1,95 +1,257 @@
+"""
+File to parse and label datafiles for the Orca project.
+
+W251 (Summer 2019) - Spyros Garyfallos, Ram Iyer, Mike Winton
+"""
+
+import numpy as np
 import os
 import pickle
+import pprint
 import random
-from collections import defaultdict
-
-import numpy as np
+import re
 import soundfile as sf
+
+from collections import defaultdict
 from sklearn.preprocessing import LabelEncoder
 
-import params
+# project-specific imports
+import orca_params
+import vggish_params as params
 
+# data path; when Docker container is run, data path on the host
+# machine is expected to be mapped to /data
+DATA_PATH = '/data/'
 
-def _quantize_sample(label, file):
+def _quantize_sample(label, file, sample_len=orca_params.FILE_MAX_SIZE_SECONDS):
+    """
+        Splits up a given file into non-overlapping segments of the specified length.
+        Returns a list containing (label, 'file:start:frames') of each segment.
+        
+        Any segments that are too short are dropped.
+    """
+    
     with sf.SoundFile(file) as wav_file:
-        # more than 1 sec
-        if wav_file.frames > wav_file.samplerate:
-            frames = int(params.file_max_size_seconds*wav_file.samplerate)
-            file_parts = np.arange(0, wav_file.frames, frames)
-            return [[label, f"{file}:{int(start)}:{frames}"] for start in file_parts]
+        # make sure sample is long enough
+        min_frames = int(sample_len * wav_file.samplerate)  # e.g. 2 * 16000
+        if wav_file.frames > min_frames:
+            file_parts = np.arange(0, wav_file.frames, min_frames)
```

Comment:
We need to make sure that we split the files in a way that never creates short segments that we afterwards remove, I think this should be done here

Reply:
Right now, this block of code first checks to make sure there's enough frames for at least one 2 sec segment (previously it was hard-coded to check for >1 sec).  Then after looping through the audio file, it drops the fractional last segment which would result in a spectogram with zero "rows".  That's the new logic.

If we want all the samples to be the same length (which I think we need to do so that all samples fed into the model are the same length), then we'll (almost) always have some straggler frames we need to drop.  The only time we wouldn't is if we get lucky and the number of frames in the file is an exact multiple of min_frames; I thought that would be unlikely enough that I just automatically dropped the last one, assuming it's too short.

Resolving the comment thread.  Please re-open if I'm missing something that we need to resolve now.

Reply:
Let me rethink this..

Reply:
We need a full training to calibrate the segment lengths, let revisit this after that